crates/storage-query-datafusion/src/table_providers.rs

```diff
@@ -233,27 +259,60 @@ where
         partition: usize,
         _context: Arc<TaskContext>,
     ) -> datafusion::common::Result<SendableRecordBatchStream> {
-        let (partition_id, partition) = self
-            .live_partitions
+        let df_partition = self
+            .logical_partitions
             .get(partition)
-            .expect("num_partitions within bounds");
-        let range = partition.key_range.clone();
-        let stream = self
-            .scanner
-            .scan_partition(
-                *partition_id,
-                range,
-                self.projected_schema.clone(),
-                self.limit,
-            )
-            .map_err(|e| DataFusionError::External(e.into()))?;
-        Ok(stream)
+            .expect("partition exists");
+
+        let scanner_factories = df_partition
+            .physical_partitions
+            .iter()
+            .map(|(partition_id, partition)| {
+                let partition_id = *partition_id;
+                let range = partition.key_range.clone();
+                let scanner = self.scanner.clone();
+                let schema = self.projected_schema.clone();
+                let limit = self.limit;
+                move || {
+                    scanner
+                        .scan_partition(partition_id, range, schema, limit)
+                        .map_err(|e| DataFusionError::External(e.into()))
+                }
+            })
+            .collect::<Vec<_>>();
+
+        let projected_schema = self.projected_schema.clone();
+        let sequential_scanners_stream = stream::iter(scanner_factories)
+            .map(move |factory| {
+                match factory() {
+                    Ok(stream) => stream,
+                    Err(err) => {
+                        // we tried to scan a partition store, but it resulted in an error.
+                        // our only option is to create a single stream that produces a single
+                        // Result::Err, the upper layer will decide what to do with that error.
+                        let error_stream = stream::once(future::ready(Err(err)));
+                        Box::pin(RecordBatchStreamAdapter::new(
```

Comment:
Not sure why we need a stream adapter here, ie to attach the schema? Its just treated as any other stream of record batches once it gets to flatten